app/models/manageiq/providers/openstack/cloud_manager.rb

```diff
@@ -54,11 +54,19 @@ class ManageIQ::Providers::Openstack::CloudManager < ManageIQ::Providers::CloudM
                 :ensure_swift_managers
 
   before_update :ensure_managers_zone_and_provider_region
+  after_save :refresh_parent_infra_provider
 
   def hostname_required?
     enabled?
   end
 
+  def refresh_parent_infra_provider
+    if provider
+      EmsRefresh.queue_refresh(provider)
```

Comment:
Is refresh on ```provider``` intentional or there should be something like ```provider.infra_manager```?

Reply:
Provider is a property on the cloud manager that refers to the infra manager when they've been attached via the provider edit form.

Reply:
At least, that's how it appeared to me. I'll double check to make sure there isn't anything funky going on with that association.

Reply:
I think it might be as @aufi  says

Reply:
You're right, we want to check if the provider was changed and then refresh `provider.infra_ems`